configure.ac

```diff
@@ -2864,8 +2864,11 @@ fi
 if test "$MPID_NO_FLOAT16" != "yes" ; then
     AC_CACHE_CHECK([whether _Float16 is supported],
     pac_cv_have_float16,[
-    AC_TRY_COMPILE(,[_Float16 a;],
-    pac_cv_have_float16=yes,pac_cv_have_float16=no)])
+    PAC_PUSH_FLAG([CFLAGS])
+    CFLAGS=""
```

Comment:
Could you add an explanation comment?

Do you think it is better to set `CFLAGS=-O0`? I am concerned that the compiler may choose a default optimization level.

Reply:
> Could you add an explanation comment?

Will do.

> Do you think it is better to set `CFLAGS=-O0`? I am concerned that the compiler may choose a default optimization level.

Possibly. MPICH's setting of the default optimization level is what got us into this mess 😆 . I'll set it to `-O0` in the next push.